crates/krator/src/runtime.rs

```diff
@@ -154,7 +153,13 @@ impl<O: Operator> OperatorRuntime<O> {
                         if meta.deletion_timestamp.is_some() {
                             reflector_deleted.notify();
                         }
-                        *reflector_manifest.write().await = manifest;
+                        match manifest_tx.broadcast(manifest) {
+                            Ok(()) => (),
+                            Err(e) => {
+                                warn!("Unable to broadcast manifest update: {:?}", e);
+                                return;
```

Comment:
I don't think it makes sense to keep trying with this task if this fails. We also haven't settled on a policy of automatically restarting the state machine in the event of an unhandled error. I'm not convinced that this would be desirable behavior. 

I could see an argument for calling `reflector_deleted.notify()` here, to signal to the other task that it needs to exit.

Reply:
I might be misunderstanding fundamentally what the operator framework is trying to achieve :)

I'll briefly describe our use case and what we would be trying to achieve with it - maybe that helps illustrate where I am coming from..

We are looking to write for example a ZookeeperOperator, this operator would watch CRDs of the type zookeepercluster and for every object of that type create some config maps and pods, which are picked up and acted upon by our Krustlet-Provider implementation. So we would have one operator and many Krustlets (PodOperator - not sure what the correct term would be).

When the operator receives a notification that the zookeepercluster object has been deleted it will delete all _dependent_ pods, causing the Krustlets to stop all services on the worker servers. So for us a delete event on the object watched by the operator is a fairly significant thing and should not be triggered by a _malfunction_ in the operator.

Not sure if that makes any sense, I am probably too close to what we are doing to explain it well. Happy to elaboate further or jump on a quick call if you think that helps!

Reply:
Not sure if I follow here or not, I don't think we are triggering a delete in the case of a return. It just stops listening for events on that state machine. I think we could maybe improve the behavior that happens in that case, but I don't think it triggers a delete based on my reread of the code

Reply:
Yeah, I'm not really sure what to do here. I can think of situations where you would not want to restart the state machine automatically (a naive implementation would recreate all the ZK pods right?). 

One thing I can suggest if the operator is only responsible for create / delete, is to mark all of the Pods, etc. as child objects of the ZookeeperCluster. K8s will then handle the garbage collection for you. I've done this with Spark.